File: benchmarks/bench_strings.js
#!/usr/bin/env node

// Concatenate "hello" 500,000 times
let s = "";
for (let i = 0; i < 500_000; i++) {
    s += "hello";
}
console.log(`concat len: ${s.length}`);

// Split a large string 100,000 times
const csv = "alpha,bravo,charlie,delta,echo,foxtrot,golf,hotel";
let total_parts = 0;
for (let j = 0; j < 100_000; j++) {
    const parts = csv.split(",");
    total_parts += parts.length;
}
console.log(`split parts: ${total_parts}`);

// Regex replace on a template string 200,000 times
const template = "Hello NAME, welcome to PLACE on DATE";
let result = "";
const pat_name = /NAME/;
const pat_place = /PLACE/;
const pat_date = /DATE/;
for (let m = 0; m < 200_000; m++) {
    result = template;
    result = result.replace(pat_name, "World");
    result = result.replace(pat_place, "Strada");
    result = result.replace(pat_date, "today");
}
console.log(`regex result: ${result}`);

File: benchmarks/bench_strings.py
#!/usr/bin/env python3

import re

# Concatenate "hello" 500,000 times
s = ""
for i in range(500_000):
    s += "hello"
print(f"concat len: {len(s)}")

# Split a large string 100,000 times
csv = "alpha,bravo,charlie,delta,echo,foxtrot,golf,hotel"
total_parts = 0
for j in range(100_000):
    parts = csv.split(",")
    total_parts += len(parts)
print(f"split parts: {total_parts}")

# Regex replace on a template string 200,000 times
template = "Hello NAME, welcome to PLACE on DATE"
result = ""
pat_name = re.compile(r"NAME")
pat_place = re.compile(r"PLACE")
pat_date = re.compile(r"DATE")
for m in range(200_000):
    result = template
    result = pat_name.sub("World", result, count=1)
    result = pat_place.sub("Strada", result, count=1)
    result = pat_date.sub("today", result, count=1)
print(f"regex result: {result}")
